esp/esdllib/esdl_script.cpp

```diff
@@ -1451,22 +1490,27 @@ class CEsdlTransformOperationTrace : public CEsdlTransformOperationWithoutChildr
                 return false;
             StringBuffer    content;
             bool            isValue;
+            ITracer& tracer = scriptContext->trace();
             if (targetContext->selectText(m_select, content, isValue))
             {
-                if (scriptContext->getTraceToStdout())
+                IDataMaskingProfileContext* masker = scriptContext->masker();
+                if (masker)
                 {
-                    if (m_label.isEmpty())
-                        printf("%s\n", content.str());
+                    if (isValue)
+                    {
+                        // TODO: need value type and optional mask style to mask a value
```

Comment:
may want to trace something saying that they selected a value, not content and it couldn't be traced because...

Reply:
Are you asking for value tracing to be blocked until the operation is updated to specify the value type? I was trying to not take away existing functionality before extending the syntax, but I can if that's what you're expecting.